### PR TITLE
Support call errors aggregation in blocks [ECR-4014]

### PR DESCRIPTION
**Pull request by dmitry-timofeev**
## Overview
Add `Blockchain#getCallErrors` and update `getTxResult` to work with the new blockchain schema in the core. The core now aggregates the call errors by blocks.

---

### ⚠️ Questions:
 - Shall we add static factory methods for `CallInBlock` (highly unclear API): https://jira.bf.local/browse/ECR-4070?
 - Shall we add extra overloads for getTxResult (eg. by location): https://jira.bf.local/browse/ECR-4014?focusedCommentId=71301&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-71301
- Matchers for errors: https://jira.bf.local/browse/ECR-4067 

---
See: https://jira.bf.local/browse/ECR-4014

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [x] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [x] Public API has Javadoc
- [x] Method preconditions are checked and documented in the Javadoc of the method
- [x] Changelog is updated if needed (in case of notable or breaking changes)
- [x] The continuous integration build passes

### CODE REVIEW

File: exonum-java-binding/core/src/main/java/com/exonum/binding/core/blockchain/Blockchain.java [hunk 1]
```diff
@@ -140,23 +151,52 @@ public long getHeight() {
...
    * @return a transaction execution result, or {@code Optional.empty()} if this transaction
    *         is unknown or was not yet executed
    */
   public Optional<ExecutionStatus> getTxResult(HashCode messageHash) {
-    MapIndex<HashCode, ExecutionStatus> txResults = getTxResults();
-    ExecutionStatus transactionResult = txResults.get(messageHash);
-    return Optional.ofNullable(transactionResult);
+    // Find tx location
+    Optional<TransactionLocation> txLocationOpt = getTxLocation(messageHash);
+    if (txLocationOpt.isPresent()) {
```
Comment (bullet-tooth) [inline exonum-java-binding/core/src/main/java/com/exonum/binding/core/blockchain/Blockchain.java]: Nit:
How about to slightly refactor this code:
1) Extract method `ExecutionStatus getExecutionStatus(TransactionLocation txLocation)`
2) Use declarative way:
```
return getTxLocation(messageHash)
.map(this::getExecutionStatus);
```

File: exonum-java-binding/core/src/main/java/com/exonum/binding/core/blockchain/Blockchain.java [hunk 2]
```diff
@@ -140,23 +151,52 @@ public long getHeight() {
   }
 
   /**
-   * Returns a map with a key-value pair of a transaction hash and execution result. Note that this
-   * is a <a href="ProofMapIndexProxy.html#key-hashing">proof map that uses non-hashed keys</a>.
+   * Returns execution errors that occurred in the block at the given height. Execution errors
```
[the remark this answers is not in the document]
Comment (dmitry-timofeev) [reply exonum-java-binding/core/src/main/java/com/exonum/binding/core/blockchain/Blockchain.java]: "Empty if no errors."

I considered to add it, but it seems redundant (we do, however, say it throws if you pass bad blockHeight).

File: exonum-java-binding/core/src/main/java/com/exonum/binding/core/blockchain/Blockchain.java (hunk 1 above)
Comment (dmitry-timofeev) [reply exonum-java-binding/core/src/main/java/com/exonum/binding/core/blockchain/Blockchain.java]: Refactored.

**Comment by dmitry-timofeev** [comment]
>    Shall we add static factory methods for CallInBlock (highly unclear API): https://jira.bf.local/browse/ECR-4070?

Yes: #1333 

>    Shall we add extra overloads for getTxResult (eg. by location): https://jira.bf.local/browse/ECR-4014?focusedCommentId=71301&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-71301

No, resolved in Jira.

**Pull request merged**